src/main/java/de/tum/cit/aet/artemis/communication/service/ReactionService.java

```diff
@@ -80,7 +92,9 @@ public Reaction createReaction(Long courseId, Reaction reaction) {
         // set user to current user
         reaction.setUser(user);
 
-        // we query the repository dependent on the type of posting and update this posting
+        // Associate and persist
+        Posting posting = reaction.getPost() != null ? reaction.getPost() : reaction.getAnswerPost();
+
```

Comment:
Why don't we directly extract the code to a method instead of writing the comment here?